Array Processing Tool/js/arrayHandler.js

```diff
@@ -0,0 +1,92 @@
+class ArrayProcessingTool{
+
+    /* O(n^2) */
+    getMaxSubSum_long(arr) {
+        let maxSum = arr[0];
+        for(let i = 0; i < arr.length; i++){
+            let tempSum = arr[i];
+            for(let j = i+1; j < arr.length; j++){
+                tempSum += arr[j];
+                if(tempSum > maxSum){
+                    maxSum = tempSum;
+                }
+            }
+            if(tempSum > maxSum){
+                maxSum = tempSum;
+            }
+        }
+        if(maxSum < 0){
+            maxSum = 0;
+        }
+        return maxSum;
+    }
+
+    /*O(n) */
+    getMaxSubSum_short(arr) {
+        let tempSum = 0, maxSum = arr[0];
+        for(let i = 0; i < arr.length; i++){
+            tempSum += arr[i];
+            if(tempSum < 0){
+                tempSum = 0;
+            }
+            if(tempSum > maxSum){
+                maxSum = tempSum;
+            }
+        }
+        return maxSum;
+    }
+
+    getMin(arr){
+        let min = arr[0];
+        for(let i = 1; i < arr.length; i++){
+            if(arr[i] < min){
+                min = arr[i];
+            }
+        }
+        return min;
+    }
+
+    getMax(arr){
+        let max = arr[0];
+        for(let i = 1; i < arr.length; i++){
+            if(arr[i] > max){
+                max = arr[i];
+            }
+        }
+        return max;
+    }
+
+    getMedium(arr){
+        let medium;
+        arr.sort(function(a,b){ 
```

Comment:
you've just changed the input parameters, so you've changed the array somewhere outside. That may leed to the bugs later